pkg/ccl/changefeedccl/scheduled_changefeed.go

```diff
@@ -0,0 +1,854 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package changefeedccl
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedpb"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedvalidators"
+	"github.com/cockroachdb/cockroach/pkg/ccl/utilccl"
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/scheduledjobs"
+	"github.com/cockroachdb/cockroach/pkg/scheduledjobs/schedulebase"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/server/telemetry"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver"
+	"github.com/cockroachdb/cockroach/pkg/sql/exprutil"
+	"github.com/cockroachdb/cockroach/pkg/sql/parser"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgnotice"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/metric"
+	"github.com/cockroachdb/errors"
+	pbtypes "github.com/gogo/protobuf/types"
+)
+
+const scheduleChangefeedOp = "CREATE SCHEDULE FOR CHANGEFEED"
+
+const (
+	optFirstRun          = "first_run"
+	optOnExecFailure     = "on_execution_failure"
+	optOnPreviousRunning = "on_previous_running"
+)
+
+var scheduledChangefeedOptionExpectValues = map[string]exprutil.KVStringOptValidate{
+	optFirstRun:          exprutil.KVStringOptRequireValue,
+	optOnExecFailure:     exprutil.KVStringOptRequireValue,
+	optOnPreviousRunning: exprutil.KVStringOptRequireValue,
+}
+
+var scheduledChangefeedHeader = colinfo.ResultColumns{
+	{Name: "schedule_id", Typ: types.Int},
+	{Name: "label", Typ: types.String},
+	{Name: "status", Typ: types.String},
+	{Name: "first_run", Typ: types.TimestampTZ},
+	{Name: "schedule", Typ: types.String},
+	{Name: "changefeed_stmt", Typ: types.String},
+}
+
+type scheduledChangefeedMetrics struct {
+	*jobs.ExecutorMetrics
+}
```

Comment:
you probably don't need this struct just to hold executorMetrics?
If, on the other hand we plan on adding more metrics, then yes, this makes sense. For now, I'd drop it.